js/listdates-min.js

```diff
@@ -1,53 +1,113 @@
 /* global moment, humanizeDuration */
+const datelistContent = document.getElementById('datelist-content');
+const datelistError = document.getElementById('datelist-error');
+const btnSortEvent = document.getElementById('btn-sort-event');
+const btnSortMonth = document.getElementById('btn-sort-month');
+const btnSortDuration = document.getElementById('btn-sort-duration');
 const xhr = new XMLHttpRequest();
+let dates;
 
-// Convert date array into HTML
-const listDates = function(dates) {
+// Sort an array by key
+// https://stackoverflow.com/a/14463464
+const sortByKey = function(array, key) {
     'use strict';
-    let content = "";
 
-    for (let i = 0; i < dates.length; i++) {
-        const eventDate = moment(dates[i].date);
+    return array.sort(function(a, b) {
+        let x = a[key];
+        let y = b[key];
+
+        if (typeof x == "string") {
+            x = x.toLowerCase();
+        }
+        if (typeof y == "string") {
+            y = y.toLowerCase();
+        }
+
+        return x < y ? -1 : x > y ? 1 : 0;
+    });
+}
```

Comment:
If you're interested in the key, why not use an object? Doesn't even need to be a fancy Class-ful object, just `let banana = {key: "value"}`

Reply:
Not sure I'm tracking you — My JSON data gives me an array of objects, and I need to sort it. Are you saying I should convert it to an object of objects? Either way, I need to get at the value of the keys in the child objects, right?

Reply:
Ah, I misunderstood what was going on. :)

Reply:
Also, I don't recall if you're using underscore in this one but it does have a beautiful little sort method: http://underscorejs.org/#sortBy which would likely be impressive to prospective employers (or at least, to me :P)